lib/shared/assert.js

```diff
@@ -0,0 +1,61 @@
+/**
+ * @fileoverview Assertion utilities equivalent to the Node.js node:asserts module.
+ * @author Josh Goldberg
+ */
+
+"use strict";
+
+/**
+ * Throws an error if the input is not truthy.
+ * @param {unknown} value The input that is checked for being truthy.
+ * @param {string} message Message to throw if the input is not truthy.
+ * @returns {void}
+ * @throws {Error}
+ */
+function ok(value, message) {
```

Comment:
```suggestion
function ok(value, message) {
```
Can we give a default message?

Reply:
We could, but that'd differ from the Node.js behavior. I feel odd forking in that way 🤔

Reply:
Oh, also, I think it'd be good to force users to provide their own precise error message, no? I can't think of a good default that would apply to all the places this is used.

Reply:
Maybe we can set `"Unspecified assertion error"` as the default message, and at some later point figure out what should be passed as a concrete message in places where the second argument is missing.

Reply:
Remember, "users" in this case is just people contributing to this code base. :smile:

Most custom `assert()` functions specify `"Assertion failed."` as the default message.

If we want to enforce a message be passed, then `assert()` should throw an error when `message` is undefined. Personally, I'm fine with a custom message as a default.

Reply:
`"Assertion failed."` default message sounds good to me.